proposals/0019-graphql.md

```diff
@@ -76,7 +81,320 @@ The result of this query is a `GraphQLResult` that encodes to the following JSON
+    let resolver = PersonResolver()
+    let schema = try! Schema<PersonResolver, NoContext> {
+        Type(Person.self) {
+            Field("name", at: \.name)
+            Field("age", at: \.age)
+            Field("height", at: \.height)
+        }
+        Mutation {
+            Field("newPerson", at: PersonResolver.newPerson) {
+                Argument("name", at: \.name)
+                Argument("age", at: \.age)
+                Argument("height", at: \.height)
+            }
+        }
+    }
+}
+```
+
+A request string for this might be:
+
+```graphql
+mutation {
+  newPerson(name: "Tyler Endicott", age: 22, height: 1.63) {
+    name
+  }
+}
+```
+
+which would generate the response:
+
+```json
+{
+  "newPerson": {
+    "name": "Tyler Endicott"
+  }
+}
+```
+
+### Input Objects
+
+Sometimes we'd like to pass a complex argument. `Input`s allow us to do this and are declared by including an `Input` block in the API declaration, composed of `InputField`s. Our example can be changed to include a mutation that creates multiple new people, each passed as an input object:
+
+```swift
+struct NewPeopleArguments: Codable {
+    let individuals: [Person]
+}
+
+struct PersonResolver {
+    func newPeople(context: NoContext, arguments: NewPeopleArguments) -> [Person] {
+        return arguments.individuals.map { person in
+            Person(
+                name: person.name,
+                age: person.age,
+                height: person.height
+            )
+        }
+    }
+}
+
+struct PersonAPI : API {
+    typealias ContextType = NoContext
+    let resolver = PersonResolver()
+    let schema = try! Schema<PersonResolver, NoContext> {
+        Type(Person.self) {
+            Field("name", at: \.name)
+            Field("age", at: \.age)
+            Field("height", at: \.height)
+        }
+        Input(Person.self) {
+            InputField("name", at: \.name)
+            InputField("age", at: \.age)
+            InputField("height", at: \.height)
+        }
+        Mutation {
+            Field("newPeople", at: PersonResolver.newPeople) {
+                Argument("individuals", at: \.individuals)
+            }
+        }
+    }
+}
+```
+
+A request might look like:
+
+```graphql
+mutation {
+  newPeople(individuals: [
+    {name: "Tyler Endicott", age: 22, height: 1.63},
+    {name: "Angelo Pappas", age: 45, height: 1.91},
+  ]) {
+    name
+  }
+}
+```
+
+which would generate the response:
+
+```json
+{
+  "newPeople": [
+    {
+      "name": "Tyler Endicott",
+      "name": "Angelo Pappas"
+    }
+  ]
+}
+```
+
+### Subscriptions
+
+Subscriptions are reactive queries that return a result whenever an event occurs. This functionality is built on Swift Concurrency using `AsyncThrowingStream`s. To create a subscription, you include a `Subscription` block in the API declaration composed of `SubscriptionFields`. We change change our API to include a subscription alert that tells the user when a 50 year storm is occurring:
```

Comment:
nit: Would maybe remove the pluralization from `AsyncThrowingStream`. I think both ways make sense but it reads cleaner to me by describing that the functionality is built on that class rather than that every subscription uses one.

Reply:
Good call. I've un-pluralized it, and reworked the sentences a bit. Thanks!